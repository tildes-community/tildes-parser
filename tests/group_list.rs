use std::fs::read_to_string;

use {insta::assert_debug_snapshot, scraper::Html, tildes_parser::GroupList};

#[test]
fn test_group_list_parsing() {
  let html = read_to_string("tests/samples/group_list.html").unwrap();
  let html = Html::parse_document(&html);
  let group_list = GroupList::from_html(&html).unwrap();
  assert_debug_snapshot!(group_list);
}
