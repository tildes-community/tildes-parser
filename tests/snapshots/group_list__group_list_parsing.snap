---
source: tests/group_list.rs
expression: group_list
---
GroupList {
    summaries: [
        GroupListSummary {
            comment_activity: Some(
                2,
            ),
            description: Some(
                "Test group with description.",
            ),
            name: "~test.description",
            topic_activity: Some(
                1,
            ),
        },
        GroupListSummary {
            comment_activity: None,
            description: None,
            name: "~test",
            topic_activity: None,
        },
    ],
}
